docs/proposals/edgemesh-design.md

```diff
@@ -76,6 +76,139 @@ Add two endpoints definition:
 |ServiceBus|Target|Request Forward|
 |Router|Source|Resutst Forward, Service Discovery, Load Balance|
 
+<img src="../images/proposals/router-design.png">
+
+* R1: EdgeController watches on k8s apiserver for Services/Endpoints.
+* R2: EC get the service definition and create router rule definition and save in DB(sqlite).
+* R3: Router gets the rule definition from DB and initializes the listeners/handlers for source and target.
+* R4: Users/Applications create a request.
+* R5: Router receives the request and forwards it to the target based on rule.
+
+Since Router fetches rules from DB, in later versions it can be started as a different process than edge-controller. Also it enables router to retrieve all the rules after process restarts.
+### Router Low-level Design
+
+### Providers
+Providers are plugins to reach a service running on edge or on cloud. For example (servicebus,eventbus running in edge_core) 
+Providers can be classified into two types
+1) Source
+2) Target
+
+In Alpha1.0 only servicebus, REST and eventbus providers will be supported.
```

Comment:
can we give some details on these providers ?

Reply:
if we can illustrate atleast one provider as an example how it will work with the router interface , just a brief description will be helpful to understand.

Reply:
I have added working of router with providers.